test/Dialect/Verif/strip-contracts.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: circt-opt --strip-contracts %s | FileCheck %s
+
+// CHECK-LABEL: func @foo
+func.func @foo(%arg0: i42) -> i42 {
+  // CHECK-NOT: verif.contract
+  %0 = verif.contract %arg0 : i42 {}
+  // CHECK: return %arg0
+  return %0 : i42
+}
+
+// CHECK-LABEL: hw.module @bar
+hw.module @bar() {
+  // CHECK-NOT: verif.contract
+  %0 = verif.contract %1 : i42 {}
+  %1 = verif.contract %0 : i42 {}
```

Comment:
What's actually the semantics if we return %1 as an output of the module? Also, the pass would still crash in that case, right? Not really important to fix here though. We have similar issues with comb.

Reply:
That's a great point, that would still crash. It would be better if it didn't, though. I think in that case we'd want to just not remove the contract. But the semantics are the same as in `%0 = comb.add %0, %0; hw.output %0` -- hard to define. Maybe some `undefined` or `poison` value, or `unknown`.

Reply:
Fixed this for now by just not removing cyclical contracts.